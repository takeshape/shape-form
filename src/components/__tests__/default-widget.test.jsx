import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ReactShallowRenderer();

import {shallow} from 'enzyme';
import DefaultWidget from '../default-widget';

function getProps(props) {
  return {
    value: 'Hello World',
    isRequired: false,
    error: '',
    config: {},
    path: 'test',
    schema: {type: 'string', title: 'Text Input'},
    onChange: jest.fn(),
    onBlur: jest.fn(),
    onFocus: jest.fn(),
    ...props
  };
}

describe('DefaultWidget', () => {
  it('renders text input', () => {
    const props = getProps();

    const tree = renderer.render(<DefaultWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders dropdown', () => {
    const props = getProps({
      value: 'foo',
      schema: {type: 'string', title: 'Text Input', enum: ['foo', 'bar', 'baz']}
    });

    const tree = renderer.render(<DefaultWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders checkbox', () => {
    const props = getProps({
      value: true,
      schema: {type: 'boolean', title: 'Checkbox Input'}
    });

    const tree = renderer.render(<DefaultWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders number input', () => {
    const props = getProps({
      value: true,
      schema: {type: 'number', title: 'Number Input'}
    });

    const tree = renderer.render(<DefaultWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('handles onChange number', done => {
    console.warn = jest.fn();
    const eventValue = '1';
    const expected = 1;

    const props = getProps({
      value: true,
      schema: {type: 'number', title: 'Number Input'},
      onChange(value) {
        expect(value).toBe(expected);
        done();
      }
    });

    const wrapper = shallow(<DefaultWidget {...props} />);
    wrapper.instance().handleChange(eventValue);
  });

  it('handles onChange number blank', done => {
    console.warn = jest.fn();
    const eventValue = '';
    const expected = undefined;

    const props = getProps({
      value: true,
      schema: {type: 'number', title: 'Number Input'},
      onChange(value) {
        expect(value).toBe(expected);
        done();
      }
    });

    const wrapper = shallow(<DefaultWidget {...props} />);
    wrapper.instance().handleChange(eventValue);
  });

  it('handles onChange string', done => {
    const eventValue = 'hello';

    const props = getProps({
      value: true,
      schema: {type: 'string', title: 'Text Input'},
      onChange(value) {
        expect(value).toBe(eventValue);
        done();
      }
    });

    const wrapper = shallow(<DefaultWidget {...props} />);
    wrapper.instance().handleChange(eventValue);
  });
});
