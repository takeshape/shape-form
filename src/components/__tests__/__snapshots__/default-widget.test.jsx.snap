// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultWidget renders checkbox 1`] = `
<CheckboxWidget
  checked={true}
  config={Object {}}
  error=""
  isRequired={false}
  onBlur={[MockFunction]}
  onChange={[Function]}
  onFocus={[MockFunction]}
  path="test"
  schema={
    Object {
      "title": "Checkbox Input",
      "type": "boolean",
    }
  }
  type="checkbox"
  value={true}
/>
`;

exports[`DefaultWidget renders dropdown 1`] = `
<TextInputWidget
  config={Object {}}
  error=""
  isRequired={false}
  onBlur={[MockFunction]}
  onChange={[Function]}
  onFocus={[MockFunction]}
  path="test"
  schema={
    Object {
      "enum": Array [
        "foo",
        "bar",
        "baz",
      ],
      "title": "Text Input",
      "type": "string",
    }
  }
  value="foo"
/>
`;

exports[`DefaultWidget renders number input 1`] = `
<TextInputWidget
  config={Object {}}
  error=""
  isRequired={false}
  onBlur={[MockFunction]}
  onChange={[Function]}
  onFocus={[MockFunction]}
  path="test"
  schema={
    Object {
      "title": "Number Input",
      "type": "number",
    }
  }
  type="number"
  value={true}
/>
`;

exports[`DefaultWidget renders text input 1`] = `
<TextInputWidget
  config={Object {}}
  error=""
  isRequired={false}
  onBlur={[MockFunction]}
  onChange={[Function]}
  onFocus={[MockFunction]}
  path="test"
  schema={
    Object {
      "title": "Text Input",
      "type": "string",
    }
  }
  type="text"
  value="Hello World"
/>
`;
